resources/providers/azurelib/auth/credentials_test.go

```diff
@@ -117,6 +121,7 @@ func TestConfigProvider_GetAzureClientConfig(t *testing.T) {
 			name: "Should return a ClientCertificateCredential",
 			config: config.AzureConfig{
 				Credentials: config.AzureClientOpt{
+					ClientCredentialsType:     config.AzureClientCredentialsTypeCertificate,
```

Comment:
I would also add tests to cover the cases where credentials type is empty / unknown value.

Reply:
The first test case has empty credentials type value.  I added one with `unknown client credentials type`.